datafs/core/data_api.py

```diff
@@ -169,6 +169,67 @@ def get_archive(self, archive_name, default_version=None):
             default_version=default_version,
             **res)
 
+    def listdir(self, location, authority_name=None):
+        '''
+        List archive path components at a given location
+
+        .. Note ::
+
+            When using listdir on versioned archives, listdir will provide the
+            version numbers when a full archive path is supplied as the
+            location argument. This is because DataFS stores the archive path
+            as a directory and the versions as the actual files when versioning
+            is on.
+
+        Parameters
+        ----------
+
+        location : str
+
+            Path of the "directory" to search
```

Comment:
Probably worthwhile to explain the convention for directories here.  "The virtual directory structure is based on the use of forward slashes (/) in the archive names."